### PR TITLE
Train glow 2m temp

### PR DESCRIPTION
**Pull request by oethier**
Training loop is functional (#4)

### CODE REVIEW

File: ift6759_train_Glow_2mTemp.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+zarr_hr = xr.open_zarr('data/processed/temp/1406/temp_1406_processed.zarr')
+
+# center it to zero
+zarr_lr, monthly_means_lr = remove_monthly_means(zarr_lr, time_dim='date')
+zarr_hr, monthly_means_hr = remove_monthly_means(zarr_hr, time_dim='date')
+
+# make data numpy arrays (unsuited for large data):
+# each ndarray have shape [date, lat, lon]
+ndarray_lr = zarr_lr.to_array().to_numpy().squeeze()[:,:16,:16]
+ndarray_hr = zarr_hr.to_array().to_numpy().squeeze()[:,:16,:16]
```
Comment (JakeColor) [inline ift6759_train_Glow_2mTemp.py]: this doesn't make sense, 16x16 arrays in hi and lo res would cover very different geographic areas... you probably did this to get the model code to work?

**Review by JakeColor — APPROVED** [review]

File: ift6759_train_Glow_2mTemp.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+# dimensionality as the high res using the nearest neighbour algo
+# We first add a channel to all images
+dataset_train_lr = dataset_train_lr.map(lambda x: x[:,:,None])
+dataset_test_lr = dataset_test_lr.map(lambda x: x[:,:,None])
+dataset_train_hr = dataset_train_hr.map(lambda x: x[:,:,None])
+dataset_test_hr = dataset_test_hr.map(lambda x: x[:,:,None])
+# Then upsample the low res datasets
+lat_hr, lon_hr = ndarray_hr.shape[1:]
+dataset_train_lr = dataset_train_lr.map(upsample(lat_hr, lon_hr, tf.image.ResizeMethod.NEAREST_NEIGHBOR))
+dataset_test_lr = dataset_test_lr.map(upsample(lat_hr, lon_hr, tf.image.ResizeMethod.NEAREST_NEIGHBOR))
```
Comment (JakeColor) [inline ift6759_train_Glow_2mTemp.py:108]: @etho44 do you understand why this is necessary? is it because the _spatial_ dimensions of the coarse data must match those of the fine?

If so, what exactly is the "nearest neighbor" algorithm doing to coarse grids? linear interpolations?

File: ift6759_train_Glow_2mTemp.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+# zipping the data together and shuffling each dataset individually
+# for "unsupervised learning"
+train_ds = preprocess_vds(dataset_train_lr, dataset_train_hr, batch_size=10, buffer_size=train_size, supervised=False)
+test_ds = preprocess_vds(dataset_test_lr, dataset_test_hr, batch_size=10, buffer_size=test_size, supervised=False)
```
Comment (JakeColor) [inline ift6759_train_Glow_2mTemp.py:113]: is this just a simple train/test split? do we not want a third test set?

Comment (oethier) [reply ift6759_train_Glow_2mTemp.py:113]: Ohh I did as they do in groenke, but you're right. This would be best practice!

File: ift6759_train_Glow_2mTemp.py (hunk 2 above)
Comment (oethier) [reply ift6759_train_Glow_2mTemp.py:108]: I must admit that I haven't checked, but I am pretty sure it's just making 4x4 data points (with the same value) from every data point in the low resolution images. But we should print some upsampled low res images to make sure.

Comment (JakeColor) [reply ift6759_train_Glow_2mTemp.py:108]: please do, i think we should understand this

File: ift6759_train_Glow_2mTemp.py (hunk 3 above)
Comment (JakeColor) [reply ift6759_train_Glow_2mTemp.py:113]: on second thought, if we're just training one model (i.e. no hyperparameter optimization), i guess it doesn't matter

Comment (oethier) [reply ift6759_train_Glow_2mTemp.py:113]: That's right
